sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -128,4 +140,15 @@ public void close() {
   public void flush(long timeoutMills) {
     // TODO: Flush transport
   }
+
+  private boolean sample() {
+    // https://docs.sentry.io/development/sdk-dev/features/#event-sampling
+    if (options.getSampling() != null && random != null) {
+      double sampling = options.getSampling();
+      if (sampling < random.nextFloat()) {
```

Comment:
if your `sampling` is `double` data type, would it not be `random.nextDouble()`?

Reply:
I'll merge it and if I am mistaken, I push a fix later on